### PR TITLE
Add security group rule with Prefix List ID to Security Groups Module

### PR DESCRIPTION
**Pull request by mattholmes15**

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
           output_path: 'results'
           platform_type: terraform
           fail_on: high,medium
-          exclude_queries: 8e94dced-9bcc-4203-8eb7-7e41202b2505 # Excludes 'Auto Scaling Group With No Associated ELB' query
+          exclude_queries: '8e94dced-9bcc-4203-8eb7-7e41202b2505,e592a0c5-5bdb-414c-9066-5dba7cdea370' # Excludes 'Auto Scaling Group With No Associated ELB' query
```
Comment (alv-lop) [inline .github/workflows/pr.yml:41]: Small thing but might be worth adding a comment for the second excluded query too.
```suggestion
          exclude_queries: '8e94dced-9bcc-4203-8eb7-7e41202b2505,e592a0c5-5bdb-414c-9066-5dba7cdea370' # Excludes 'Auto Scaling Group With No Associated ELB' and 'IAM Access Analyzer Undefined' queries
```

Comment (mattholmes15) [reply .github/workflows/pr.yml]: Agreed, updated

File: modules/database/variables.tf [hunk 2]
```diff
@@ -5,13 +5,18 @@ variable "allowed_security_groups" {
...
   type = object({
     username = string,
     password = string,
   })
 }
 
+variable "kms_key_policy" {
+  description = "Custom Key Policy for the Aurora KMS Key"
+  type        = string
+  default     = ""
```
Comment (dan-hill2802) [inline modules/database/variables.tf:17]: would `null` be more reliable as a default?

Comment (mattholmes15) [reply modules/database/variables.tf]: Yeah I think so, I now remember that we were moving away from empty quotes and using `null` a while back, updated

**Review by alv-lop — APPROVED** [review]

**Review by dan-hill2802 — APPROVED** [review]

**Pull request merged**